.github/workflows/cre-local-env-tests.yaml

```diff
@@ -42,9 +45,28 @@ jobs:
           ref: ${{ github.event_name == 'pull_request' && github.sha || inputs.chainlink_version }}
 
       - name: Set up Go
+        id: setup-go
         uses: actions/setup-go@v5
         with:
           go-version-file: core/scripts/go.mod
+          cache: true
+
+      - name: Download Go dependencies
+        id: go-deps
+        run: |
+          cd core/scripts/cre/environment
```

Comment:
Just a nit but you can use [working-directory: ...](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsworking-directory) to avoid `cd` in the runs.